server/api/src/main/java/org/hawkular/apm/server/api/task/ProcessingUnit.java

```diff
@@ -106,7 +106,12 @@ public void handle(String tenantId, List<T> items) throws Exception {
         List<T> retries = null;
         RetryAttemptException lastException = null;
 
-        processor.initialise(tenantId, items);
+        try {
+            processor.initialise(tenantId, items);
+        } catch (RetryAttemptException e) {
+            retries = items;
```

Comment:
It doesn't really need to - whether the initialisation fails, or the processing of all the events (as a batch) fails, the end result is that the events need to be retried.